src/varlink.js

```diff
@@ -80,7 +80,9 @@ function connect(address) {
         channel.close();
     };
 
-    return connection;
+    return new Promise((resolve, reject) => {
+        channel.addEventListener("ready", () => resolve(connection));
```

Comment:
In order for a client to actually detect if the connection failed, should the `close` signal be wired to `reject()`? If one gets a "close" before a "ready", then there's surely something wrong, and the `ready` part could then just remove the event listener for `close`.

Reply:
Also, doesn't this need to remove its own addEventListener(), otherwise we'd forever "leak" this listener?

Reply:
Indeed. Fixed.